gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/EntityServiceAndGatewayServiceConverter.java

```diff
@@ -205,6 +197,34 @@ private static Function convertToQueryFunction(
         {
           builder.setFunctionName(function.getFunction().name()).setAlias(function.getAlias());
 
+          // Backward compatibility to handle long values
+          if (function.getFunction() == FunctionType.AVGRATE) {
+            List<org.hypertrace.gateway.service.v1.common.Expression> columns =
+                function.getArgumentsList().stream()
+                    .filter(
+                        e ->
+                            e.hasLiteral()
+                                && e.getLiteral().getValue().getValueType()
+                                    == org.hypertrace.gateway.service.v1.common.ValueType.LONG)
+                    .collect(Collectors.toList());
+
+            if (columns.size() > 0) {
+              columns.forEach(
+                  e ->
+                      builder.addArguments(
+                          Expression.newBuilder()
+                              .setLiteral(
+                                  LiteralConstant.newBuilder()
+                                      .setValue(
+                                          Value.newBuilder()
+                                              .setString(longToIso(e))
```

Comment:
we can inline function `longToIso` with `Duration.ofSeconds(expression.getLiteral().getValue().getLong()).toString()`